src/lib/features/context/context-service.ts

```diff
@@ -140,6 +140,45 @@ class ContextService {
         });
     }
 
+    async updateContextFieldLegalValue(
+        contextFieldLegalValue: { name: string; legalValue: LegalValueSchema },
+        auditUser: IAuditUser,
+    ): Promise<void> {
+        const contextField = await this.contextFieldStore.get(
+            contextFieldLegalValue.name,
+        );
+        const validatedLegalValue = await legalValueSchema.validateAsync(
+            contextFieldLegalValue.legalValue,
+        );
+
+        const legalValues = contextField.legalValues
+            ? [...contextField.legalValues]
+            : [];
+
+        const existingIndex = legalValues.findIndex(
+            (legalvalue) => legalvalue.value === validatedLegalValue.value,
+        );
```

Comment:
Oh, my. we store this in a list, huh? I imagine using a map would be better, but it's probably also not something we want to touch now, I guess 😅

Reply:
But then again, maybe order is important? 🤷🏼

Reply:
Yes, this PR piggybacks on the current implementation